### PR TITLE
refactor(pie-link): DSW-2280 use classes instead of attributes for styling

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Describe your changes (can list changeset entries if preferable)
- Use classes instead of attributes for styling pie-link.
- Update component generator to reduce the number of `type` changes in future.

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have reviewed the `PIE Storybook` PR preview
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have used the `/snapit` functionality to test my changes in a consuming application

### Reviewer 1 - @jamieomaguire 
- [x] I have reviewed the `PIE Storybook` PR preview

### Reviewer 2
- [x] I have reviewed the `PIE Storybook` PR preview

### CODE REVIEW

File: packages/components/pie-link/test/visual/pie-link.spec.ts [hunk 1]
```diff
@@ -51,7 +50,7 @@ test.beforeEach(async ({ mount }, testInfo) => {
 
 componentVariants.forEach((variant) => test(`should render all prop variations for Variant: ${variant}`, async ({ page, mount }) => {
     for (const combo of componentPropsMatrixByVariant[variant]) {
-        const testComponent: WebComponentTestInput = createTestWebComponent(combo, renderTestPieLink);
```
Comment (fernandofranca) [inline packages/components/pie-link/test/visual/pie-link.spec.ts:54]: Why this one was changed?

**Review by fernandofranca — APPROVED** [review]
Look great! 
Left a single question.

File: packages/components/pie-link/test/visual/pie-link.spec.ts (hunk 1 above)
Comment (xander-marjoram) [reply packages/components/pie-link/test/visual/pie-link.spec.ts:54]: Hey, the type is already inferred from the return type of the function, so I think adding it here was causing some minor clutter.

Comment (xander-marjoram) [reply packages/components/pie-link/test/visual/pie-link.spec.ts:54]: @fernandofranca Hey, the type is already being inferred from the return type of the function, so I think duplicating it here was causing some minor clutter.

**Pull request merged**